src/AlgoliaSearchCore.js

```diff
@@ -438,6 +421,52 @@ AlgoliaSearchCore.prototype._jsonRequest = function(initialOpts) {
     }
   }
 
+  function interopCallbackReturn(promise, callback) {
+    // either we have a callback
+    // either we are using promises
+    if (typeof initialOpts.callback === 'function') {
+      promise.then(function okCb(content) {
+        exitPromise(function() {
+          initialOpts.callback(null, callback(content));
+        }, client._setTimeout || setTimeout);
+      }, function nookCb(err) {
+        exitPromise(function() {
+          initialOpts.callback(err);
+        }, client._setTimeout || setTimeout);
+      });
+    } else {
+      return promise.then(function(content) {
```

Comment:
Why isn't that `promise.then(callback)`?

Reply:
For no particular reason, I'll update the call to the function. I'll also rename the parameters of the function `interopCallbackReturn` to be a bit more explicit.